Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityDeviceManager.cs

```diff
@@ -366,23 +363,33 @@ public override void Enable()
             //NOTE: We update the source state data, in case an app wants to query it on source detected.
             for (var i = 0; i < numInteractionManagerStates; i++)
             {
-                var controller = GetController(interactionManagerStates[i].source);
-
-                if (controller != null)
-                {
-                    controller.UpdateController(interactionManagerStates[i]);
-                    inputSystem?.RaiseSourceDetected(controller.InputSource, controller);
-                }
+                CreateController(interactionManagerStates[i]);
             }
 
-            if ((inputSystem != null) &&
-                InputSystemProfile.GesturesProfile != null &&
+            if (InputSystemProfile.GesturesProfile != null &&
                 InputSystemProfile.GesturesProfile.WindowsGestureAutoStart == AutoStartBehavior.AutoStart)
             {
                 GestureRecognizerEnabled = true;
             }
         }
 
+        private void CreateController(InteractionSourceState interactionSourceState)
+        {
+            var controller = GetController(interactionSourceState.source);
+
+            if (controller != null)
+            {
+                var raiseSourceDetected = !activeControllers.ContainsKey(interactionSourceState.source.id);
```

Comment:
nit: Can you move this line down next to the if statement so they are more local. Also can you change the var to a bool. Generally var is useful for a class or interface type especially for like Dictionary<IMIxedRelaitySErvice, List<IMixedRelaityController>> blah things. But it's clearer with it being a bool in this case

Also maybe add a comment line like "If this is a new detected controller, raise source detected event with input system"

Thanks!

Reply:
just recognized that this wasn't quite right. Needed to change order of code and put some additional comments.
Can you check it out again?